nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/FlowResource.java

```diff
@@ -3039,6 +3098,86 @@ public Response getTemplates() {
         return generateOkResponse(entity).build();
     }
 
+    // -------------
+    // flow-analysis
+    // -------------
+
+    /**
+     * Returns flow analysis results produced by the analysis of a given process group.
+     *
+     * @return a flowAnalysisResultEntity containing flow analysis results produced by the analysis of the given process group
+     */
+    @GET
+    @Consumes(MediaType.WILDCARD)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("flow-analysis/result/{processGroupId}")
+    @ApiOperation(
+        value = "Returns flow analysis results produced by the analysis of a given process group",
+        response = FlowAnalysisResultEntity.class,
+        authorizations = {
+            @Authorization(value = "Read - /controller")
+        })
+    @ApiResponses(value = {
+        @ApiResponse(code = 400, message = "NiFi was unable to complete the request because it was invalid. The request should not be retried without modification."),
+        @ApiResponse(code = 401, message = "Client could not be authenticated."),
+        @ApiResponse(code = 403, message = "Client is not authorized to make this request."),
+        @ApiResponse(code = 404, message = "The specified resource could not be found."),
+        @ApiResponse(code = 409, message = "The request was valid but NiFi was not in the appropriate state to process it. Retrying the same request later may be successful.")
+    })
+    public Response getFlowAnalysisResults(
+        @Context final HttpServletRequest httpServletRequest,
+        @ApiParam(
+            value = "The id of the process group representing (a part of) the flow to be analyzed.",
+            required = true
+        )
+        @PathParam("processGroupId")
+        final String processGroupId
+    ) {
+        if (isReplicateRequest()) {
+            return replicate(HttpMethod.GET);
+        }
+
+        authorizeFlow();
+
+        FlowAnalysisResultEntity entity = serviceFacade.getFlowAnalysisResult(processGroupId);
+
+        return generateOkResponse(entity).build();
+    }
+
+    /**
+     * Returns all flow analysis results currently in effect.
+     *
+     * @return a flowAnalysisRuleEntity containing all flow analysis results currently in-effect
+     */
+    @GET
+    @Consumes(MediaType.WILDCARD)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("flow-analysis/result")
```

Comment:
It seems like this path should be pluralized to `/results`:
```suggestion
    @Path("flow-analysis/results")
```